app/models/email_message.rb

```diff
@@ -11,4 +11,23 @@ def self.find_for_reports(feedback_message_ids)
     select(:to, :subject, :content, :utm_campaign, :feedback_message_id)
       .where(feedback_message_id: feedback_message_ids)
   end
+
+  def self.fast_destroy_old_retained_email_deliveries(destroy_before_timestamp = 3.months.ago)
+    # We remove email delivery records periodically, except some we retain long term.
+    # We generally want to retain emails directly sent by human admins.
+    # The only email currently sent manually are those that are tied directly to a feedback message.
+    sql = <<-SQL.squish
+      DELETE FROM ahoy_messages
+      WHERE ahoy_messages.id IN (
+        SELECT ahoy_messages.id
+        FROM ahoy_messages
+        WHERE created_at < ? AND feedback_message_id = null
```

Comment:
> feedback_message_id = null

💡

Reply:
Yup, definitely. There are a few ways we could do this. My hope is that if we need to evolve this logic later, it's encapsulated enough that we can go with anything that works.